### PR TITLE
Initializes conhost's Campbell color scheme in conhost order instead of ANSI/VT order

### LINKED ISSUE #1234: conhost initializes campell color scheme in ANSI/VT order instead of conhost order
From MSFT:21086112

do a vtpipeterm without the headless. The colors are in the wrong order.

_InitializeCampbellScheme does it in the linux order, not the windows order. There needs to be an initializeWindows and initializeTerminal version of those functions.
**People without settings will get the totally wrong colors, because we'll initialize the table to the wrong values**

image

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request
Switch conhost back to BGR order instead of RGB order.

## PR Checklist
* [x] Closes #1234 

## Validation Steps Performed
I ran `openvt`, as shown in the original bug

### CODE REVIEW

**Comment by miniksa** [comment]
> I am wondering if it is reasonable to have a function called `SwapANSIColorOrderForConhost` that just works on a vector, and all of our `Initialize...Color` functions use ANSI order?

I would very much prefer this to a second definition of the entire table.

**Comment by miniksa** [comment]
@zadjii-msft, can you fix this up before the bot kills it?

File: src/types/utils.cpp [hunk 1]
```diff
@@ -146,6 +147,29 @@ void Utils::InitializeCampbellColorTable(gsl::span<COLORREF>& table)
...
+// Return Value:
+// - <none>, throws if the table has less that 16 entries
+void Utils::InitializeCampbellColorTableForConhost(gsl::span<COLORREF>& table)
+{
+    THROW_HR_IF(E_INVALIDARG, table.size() < 16);
+    InitializeCampbellColorTable(table);
+    SwapANSIColorOrderForConhost(table);
+}
+
+void Utils::SwapANSIColorOrderForConhost(gsl::span<COLORREF>& table)
```
Comment (miniksa) [inline src/types/utils.cpp]: Doc comment.

Comment (miniksa) [reply src/types/utils.cpp]: And technically probably unit test...

Comment (DHowett-MSFT) [reply src/types/utils.cpp]: ooh make a cool ascii art diagram about why you've gotta do

**Pull request merged**